src/hugr/views/sibling_subgraph.rs

```diff
@@ -517,16 +515,34 @@ fn validate_subgraph<H: HugrView>(
     }
 
     let mut ports_inside = inputs.iter().flatten().chain(outputs).copied();
-    let mut ports_outside = ports_inside
-        .clone()
-        .flat_map(|(n, p)| hugr.linked_ports(n, p));
     // Check incoming & outgoing ports have target resp. source inside
     let nodes = nodes.iter().copied().collect::<HashSet<_>>();
     if ports_inside.any(|(n, _)| !nodes.contains(&n)) {
         return Err(InvalidSubgraph::InvalidBoundary);
     }
-    // Check incoming & outgoing ports have source resp. target outside
-    if ports_outside.any(|(n, _)| nodes.contains(&n)) {
+    // Check that every inside port has at least one linked port outside.
+    if ports_inside.any(|(n, p)| hugr.linked_ports(n, p).all(|(n1, _)| nodes.contains(&n1))) {
+        return Err(InvalidSubgraph::NotConvex);
```

Comment:
I notice that my use of the notion of convexity is a bit misleading here. Maybe replacing the `NotConvex` error with `InvalidBoundary` would be more appropriate. The function could also be renamed to e.g. `validate_boundary_ports`.

Feel free to choose whatever you find clearer.